### ISSUE TITLE
[Proposal] Implement JsonSerializable Interface 

### ISSUE THREAD

**Issue by Domi04151309** [uestla/Simplex-Calculator#20]
It would be nice if the result could be converted to JSON by having the classes implement [`JsonSerializable`](https://www.php.net/manual/en/jsonserializable.jsonserialize.php).

**Comment by uestla** [comment]
The result can be simply just by `json_encode($solver->getSolution())`

**Comment by Domi04151309** [comment]
Yes, that's right. What I meant was being able to call `json_encode($solver)` to get a JSON-Object containing all public variables  (accessible through the getters) in a similar format to `print_r`.
